code/game/objects/items/gift.dm

```diff
@@ -30,6 +33,17 @@
 		to_chat(M, "<span class='warning'>You're supposed to be spreading gifts, not opening them yourself!</span>")
 		return
 
+	var/gift_type = get_gift_type()
+
+	if(!ispath(gift_type, /obj/item))
+		return
```

Comment:
excessive sanity checking, unless you're intentionally trying to stop badmins from making the gifts list only contain bears or something.